app/models/armor.rb

```diff
@@ -80,39 +35,38 @@ def tempering_materials
 
   private
 
-  def set_canonical_armor
-    canonicals = canonical_models
+  def canonical_class
+    Canonical::Armor
+  end
 
-    unless canonicals.count == 1
-      clear_canonical_armor
-      return
-    end
+  def canonical_table
+    'canonical_armors'
```

Comment:
This could have been extracted as a constant, but this way we can define this in the superclass with a `NotImplementedError` and more easily find if this method isn't defined.